pkg/services/notification/resource_control/address.go

```diff
@@ -41,29 +42,117 @@ func DescribeAddresses(ctx context.Context, req *pb.DescribeAddressesRequest) ([
 	offset := pbutil.GetOffsetFromRequest(req)
 	limit := pbutil.GetLimitFromRequest(req)
 
-	var addrs []*models.Address
+	if len(req.AddressListId) == 0 {
+		var addrs []*models.Address
+		var count uint64
+
+		if err := nfdb.GetChain(global.GetInstance().GetDB().Table(models.TableAddress)).
+			AddQueryOrderDir(req, models.AddrColCreateTime).
+			BuildFilterConditions(req, models.TableAddress).
+			Offset(offset).
+			Limit(limit).
+			Find(&addrs).Error; err != nil {
+			logger.Errorf(ctx, "Describe address failed, %+v.", err)
+			return nil, 0, err
+		}
+
+		if err := nfdb.GetChain(global.GetInstance().GetDB().Table(models.TableAddress)).
+			BuildFilterConditions(req, models.TableAddress).
+			Count(&count).Error; err != nil {
+			logger.Errorf(ctx, "Describe address count failed, %+v.", err)
+			return nil, 0, err
+		}
+
+		return addrs, count, nil
+	} else {
+		return nil, 0, nil
+	}
+}
+
+func DescribeAddressesWithAddrListId(ctx context.Context, req *pb.DescribeAddressesRequest) ([]*models.AddressWithListId, uint64, error) {
+	dbChain := nfdb.GetChain(global.GetInstance().GetDB().Table("address t1").
+		Select("t1.*,t3.address_list_id").
+		Joins("left join address_list_binding t2 on t1.address_id=t2.address_id").
+		Joins("left join address_list t3 on t2.address_list_id=t3.address_list_id"))
+
+	offset := pbutil.GetOffsetFromRequest(req)
+	limit := pbutil.GetLimitFromRequest(req)
+
+	var addrs []*models.AddressWithListId
 	var count uint64
 
-	if err := nfdb.GetChain(global.GetInstance().GetDB().Table(models.TableAddress)).
-		AddQueryOrderDir(req, models.AddrColCreateTime).
-		BuildFilterConditions(req, models.TableAddress).
+	dbChain, orderByStr := buildDB4DescAddrsWithListId(dbChain, req)
+
+	err := dbChain.
 		Offset(offset).
 		Limit(limit).
-		Find(&addrs).Error; err != nil {
-		logger.Errorf(ctx, "Describe address failed, %+v.", err)
+		Order(orderByStr).
+		Scan(&addrs).
+		Error
+	if err != nil {
+		logger.Errorf(nil, "Failed to Describe Addresses [%v], error: %+v.", req, err)
 		return nil, 0, err
 	}
 
-	if err := nfdb.GetChain(global.GetInstance().GetDB().Table(models.TableAddress)).
-		BuildFilterConditions(req, models.TableAddress).
-		Count(&count).Error; err != nil {
-		logger.Errorf(ctx, "Describe address count failed, %+v.", err)
+	err = dbChain.Count(&count).Error
+	if err != nil {
+		logger.Errorf(nil, "Failed to Describe Addresses [%v], error: %+v.", req, err)
 		return nil, 0, err
 	}
 
 	return addrs, count, nil
 }
 
+func buildDB4DescAddrsWithListId(dbChain *nfdb.Chain, req *pb.DescribeAddressesRequest) (*nfdb.Chain, string) {
```

Comment:
Just need a prefix?  
You can add prefix as a parameter to func BuildFilterConditions to support this.